### PR TITLE
IP Block List

### PR DESCRIPTION
**Pull request by aerosol**
### Changes

This PR introduces a basic IP Block List feature. Users are able to add up to 30 IP (v4/v6) Addresses that will be rejected on ingest. A common use case is to block own IP, and this is hopefully made easy. IP ranges (netmask support) will be implemented in a follow-up if there's demand.

The rules are kept in a separate cache process for ingestion lookups and refreshed periodically.
BTW some room is already made for the upcoming Shields - the main live view will later on contain tabs with rule counters.

Reviewing commit-by-commit should work, and apologies for +1k, can split if anyone's struggling.

The docs will require an update + the ℹ️ heroicon link is currently missing from the panel. Migration will be extracted to a separate PR once review is conducted.

~~There's one minor annoyance with `Ecto.Network`, ref https://github.com/adam12/ecto_network/pull/28~~

https://github.com/plausible/analytics/assets/173738/51503f91-96e4-4e10-84cb-d6d1c9cb81d1

edit: Added simple "updated_by" tracking

https://github.com/plausible/analytics/assets/173738/0d2572fb-750c-46ec-8ae6-8b3540ebaddf

### Tests
- [x] Automated tests have been added
- [ ] This PR does not require tests

### Changelog
- [x] Entry has been added to changelog
- [ ] This PR does not make a user-facing change

### Documentation
- [ ] [Docs](https://github.com/plausible/docs) have been updated
- [ ] This change does not need a documentation update

### Dark mode
- [x] The UI has been tested both in dark and light mode
- [ ] This PR does not change the UI

### CODE REVIEW

**Review by ukutaht — APPROVED** [review]
Couple minor comments about naming but otherwise looks great!

File: lib/plausible/ingestion/event.ex [hunk 1]
```diff
@@ -151,6 +153,19 @@ defmodule Plausible.Ingestion.Event do
     end
   end
 
+  defp drop_blocklist_ip(%__MODULE__{} = event) do
```
[the remark this answers is not in the document]
Comment (ukutaht) [reply lib/plausible/ingestion/event.ex:156]: Alternatively - use the same naming scheme as for the Postgres tables: `drop_shield_rule_ip`, `drop_shield_rule_country`, etc.

File: lib/plausible/shield/ip_rule.ex [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  @type t() :: %__MODULE__{}
+
+  @primary_key {:id, :binary_id, autogenerate: true}
+  schema "shield_rules_ip" do
+    belongs_to :site, Plausible.Site
+    field :inet, EctoNetwork.INET
+    field :action, Ecto.Enum, values: [:deny, :allow], default: :deny
+    field :description, :string
+    field :updated_by, :string
```
[the remark this answers is not in the document]
Comment (aerosol) [reply lib/plausible/shield/ip_rule.ex]: https://github.com/plausible/analytics/pull/3761/commits/2099fa563041f96626fe00cb7b29d6c970e6bb06

File: lib/plausible/ingestion/event.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (aerosol) [reply lib/plausible/ingestion/event.ex]: https://github.com/plausible/analytics/pull/3761/commits/c4d3a49be750e39cb8fc1ab259b28869b7165eab

**Pull request merged**